### PR TITLE
feat(serve): embed user hash in preview URLs for abuse tracing

### PR DESCRIPTION
**Pull request by catalan-adobe**
## Summary

- Adds a non-reversible 8-hex-char user hash to every `serve` preview URL, enabling tracing of who published content on `sliccy.now`/`sliccy.dev`
- Hash is `SHA-256(providerId:userName)` truncated to 4 bytes — same user always produces the same hash; verify a suspect by recomputing
- Also stored in `PreviewRecord.userHash` for server-side DO lookup

**New URL format** (exactly 63 chars, the DNS label limit):
```
<compactUUID>--<userHash8>-<secret20>.sliccy.now
```
The `-` at position 8 of the post-separator segment discriminates new from old format; old URLs resolve without change.

**Secret** reduced from 12 → 10 bytes (2^80, still astronomically unguessable).

**Identity source:** first logged-in provider account, priority `adobe > github > any`, with `00000000` for anonymous users.

## Test plan

- [ ] `npx vitest run packages/shared-ts/tests/preview-url.test.ts` — new format encoding cases
- [ ] `npx vitest run packages/cloudflare-worker/tests/preview-host.test.ts` — new/old format parsing, backward compat
- [ ] `npx vitest run packages/cloudflare-worker/tests/session-tray-preview.test.ts` — userHash stored in PreviewRecord
- [ ] `npm run typecheck` — all 8 targets clean
- [ ] `npm run build -w @slicc/cloudflare-worker` — worker builds and dry-run passes

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: packages/webapp/src/ui/boot/setup-standalone-panel-rpc.ts [hunk 1]
```diff
@@ -11,11 +11,26 @@
...
+  const accounts = getAccounts();
+  const priority = ['adobe', 'github'];
+  const account =
+    priority
+      .flatMap((id) => accounts.filter((a) => a.providerId === id && a.userName && !a.loggedOut))
+      .find(Boolean) ?? accounts.find((a) => a.userName && !a.loggedOut);
+  if (!account?.userName) return '00000000';
+  const input = `${account.providerId}:${account.userName}`;
+  const bytes = await crypto.subtle.digest('SHA-256', new TextEncoder().encode(input));
+  return Array.from(new Uint8Array(bytes, 0, 4), (b) => b.toString(16).padStart(2, '0')).join('');
```
[the remark this answers is not in the document]
Comment (catalan-adobe) [reply packages/webapp/src/ui/boot/setup-standalone-panel-rpc.ts]: Fixed in be8ed158. `computeUserHash` is now wrapped in try/catch — any error (including a `crypto.subtle` rejection) returns `'00000000'` so the serve mint continues. The server-side shape validation is in the follow-up commit (400 on non-8-hex-char `userHash` in `preview-routes.ts`).

File: packages/webapp/src/ui/boot/setup-standalone-panel-rpc.ts [hunk 2]
```diff
@@ -11,11 +11,26 @@
...
 import type { RemoteCdpPageBridge } from '../remote-cdp-page-bridge.js';
 
+/** SHA-256(providerId:userName) truncated to 8 hex chars. Returns '00000000' for anonymous. */
+async function computeUserHash(): Promise<string> {
+  const accounts = getAccounts();
+  const priority = ['adobe', 'github'];
+  const account =
+    priority
+      .flatMap((id) => accounts.filter((a) => a.providerId === id && a.userName && !a.loggedOut))
+      .find(Boolean) ?? accounts.find((a) => a.userName && !a.loggedOut);
```
[the remark this answers is not in the document]
Comment (catalan-adobe) [reply packages/webapp/src/ui/boot/setup-standalone-panel-rpc.ts]: Fixed in be8ed158 — switched to the priority-map pattern: `['adobe','github'].map(id => active.find(a => a.providerId === id)).find(Boolean) ?? active[0]`. Clearer intent, avoids the double-filter flatMap.

File: packages/cloudflare-worker/src/preview-host.ts [hunk 3]
```diff
@@ -44,7 +50,16 @@ export function previewTokenFromHost(host: string): string | null {
...
+
+  // Discriminate new format (userHash8-secret) from old (pure hex secret).
+  // New: remainder[8] === '-'; old: remainder is pure hex (no '-').
+  if (remainder.length > 8 && remainder[8] === '-') {
+    const userHash = remainder.slice(0, 8);
+    const secret = remainder.slice(9);
+    return { token: `${rehyphenateUuid(compactUuid)}.${secret}`, userHash };
+  }
+
+  return { token: `${rehyphenateUuid(compactUuid)}.${remainder}`, userHash: null };
```
[the remark this answers is not in the document]
Comment (catalan-adobe) [reply packages/cloudflare-worker/src/preview-host.ts:64]: The invariant holds: `createCapabilityToken` produces pure hex (no `-`), so old-format secrets can never trigger the new-format discriminator. The validation guard added to `preview-routes.ts` (400 on non-`[0-9a-f]{8}`) enforces this at the HTTP boundary so no future caller can produce a hash containing `-` either. No code change needed here, but the edge cases are now covered by four new tests added in 0b236dcc.

File: packages/cloudflare-worker/src/preview-routes.ts [hunk 4]
```diff
@@ -62,6 +64,7 @@ export async function handlePreviewMint(request: Request, trayStub: TrayStub): P
         bridge: body.bridge,
         maxTabs: body.maxTabs,
         webhookId: body.webhookId,
+        userHash: body.userHash,
```
[the remark this answers is not in the document]
Comment (catalan-adobe) [reply packages/cloudflare-worker/src/preview-routes.ts:70]: Fixed in be8ed158: `preview-routes.ts` now rejects any `userHash` that isn't exactly `[0-9a-f]{8}` with a 400 before forwarding to the DO.

File: packages/cloudflare-worker/src/session-tray-preview.ts [hunk 5]
```diff
@@ -401,7 +402,7 @@ export async function mintPreview(
   const { createCapabilityToken } = await import('./shared.js');
   const { buildPreviewUrl } = await import('@slicc/shared-ts');
 
-  const previewToken = createCapabilityToken(tray.trayId, 12);
+  const previewToken = createCapabilityToken(tray.trayId, 10);
```
[the remark this answers is not in the document]
Comment (catalan-adobe) [reply packages/cloudflare-worker/src/session-tray-preview.ts:405]: `handlePreviewList` already returns the full `PreviewRecord[]` verbatim, so `userHash` is present in the list response from the DO today. The gap was on the client side — `PreviewListItem` in `preview-mint-client.ts` didn't declare the field. Fixed in be8ed158: `userHash?` is now part of `PreviewListItem`.

File: packages/cloudflare-worker/tests/preview-host.test.ts [hunk 6]
```diff
@@ -50,4 +59,35 @@ describe('previewTokenFromHost', () => {
...
+      token: '1ed1735c-dce7-43d8-9e88-3a547d5deb5a.aabbccddeeff00112233',
+      userHash: 'deadbeef',
+    });
+  });
+
+  it('extracts token and userHash from new-format localhost host', () => {
+    const result = previewTokenFromHost(
+      '550e8400e29b41d4a716446655440000--00000000-ff00112233445566778899aa.localhost:8787'
+    );
+    expect(result).toEqual({
```
[the remark this answers is not in the document]
Comment (catalan-adobe) [reply packages/cloudflare-worker/tests/preview-host.test.ts:88]: Added all four cases in 0b236dcc: DNS 63-char limit, 8-char pure-hex (old format), 7-char prefix (old format), and empty-secret new format (parses without crashing, DO lookup 404s). 14 tests pass.
